server/cmwell-bg/src/main/scala/cmwell/bg/CMWellBGActor.scala

```diff
@@ -177,19 +179,23 @@ class CMWellBGActor(partition:Int, config:Config, irwService:IRWService, ftsServ
   }
 
   private def startImp = {
-    if(impStream == null) {
-      logger info "starting ImpStream"
-      impStream = new ImpStream(partition, config, irwService, zStore, ftsService, offsetsService, self)
-    } else
-      logger warn "requested to start Imp Stream but it is already running. doing nothing."
+    if(impOn) {
+      if (impStream == null) {
+        logger info "starting ImpStream"
+        impStream = new ImpStream(partition, config, irwService, zStore, ftsService, offsetsService, self)
+      } else
+        logger warn "requested to start Imp Stream but it is already running. doing nothing."
+    }
   }
 
   private def startIndexer = {
```

Comment:
same for `startIndexer` better have a `lazy val` instead `var` + `private def` with nullity check

Reply:
same as above